### DISCUSSION TITLE
How to output text and char to JSON file at the same time?

### DISCUSSION THREAD

**Discussion by trulymust** [pymupdf/PyMuPDF#1785] [Q&A] [answered]
Hello,developer，I now have a requirement to output text and char to JSON files at the same time. but,Your document shows that this operation is not supported. I can only manually implement it later?
![98bf1038b53c9bb3053eae0ed8ad1b0](https://user-images.githubusercontent.com/55483183/176070765-c9ff459b-a211-4f7b-91db-3f41211b0f64.png)
My idea is to add char span under text span to display the position and content of single characters in each line.
Thank you!

**Comment by JorjMcKie** [comment, marked as answer]
The two variants (generated by "dict" and "rawdict", respectively) differ in the span level only.
In the first case case, the span has the key `"text"`, the rawdict spans have `"chars"` **_instead_**.

So you could generate "rawdict" output, then walk through each of its spans, concatenate all `char["c"]` of its `"chars"` and put the result in the span as `"text"` key.

```python
for b in page.get_text("rawdict", flags=..., clip=..., etc.)["blocks"]:
    for l in b["lines"]:
        for s in l["spans"]:
            s["text"] = "".join([c["c"] for c in s["chars"]])
```

**Reply by trulymust** [reply]
Thanks for your reply, Best wishees for you

**Comment by JorjMcKie** [comment]
To output to a JSON file, you could even write a small JSON output plugin which does the above.
That would reduce the additional memory load and processing time to a minimum.
